pkg/appctx/container.go

```diff
@@ -4,11 +4,16 @@ import (
 	"context"
 	"fmt"
 	"sync"
+
+	"github.com/justtrackio/gosoline/pkg/conc"
 )
 
 type containerKeyType int
 
-var containerKey containerKeyType = 1
+var (
+	containerKey containerKeyType = 1
+	lock                          = conc.NewKeyLock()
```

Comment:
The lock shouldn't be global but bound to the container, right? :thinking:

Reply:
hmm the container itself uses sync.Map so that should not be a problem. But I tried removing the lock completely and I can't seem to reproduce the race condition I had in the past.

Reply:
but you are right there seem to not be a reason to have a global lock at this place

Reply:
ok scratch the comment before :D , so the issue does not occur anymore because of the changes I did to the metric factory.  So it doesn't add 2 modules 1 for Prometheus API server and one for the Metric daemon. But it could be in some edge cases that we'll need the lock, so I'll add it to the container

Reply:
:+1: